libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m

```diff
@@ -602,7 +597,7 @@ - (void)passcodeValidatedToAuthValidation
 
 - (void)authValidationAtLaunch
 {
-    if (![SFUserAccountManager sharedInstance].isCurrentUserAnonymous && ![SFUserAccountManager sharedInstance].currentUser.credentials.accessToken && self.authenticateAtLaunch) {
+    if (self.authenticateAtLaunch && ([SFUserAccountManager sharedInstance].currentUser==nil || ![SFUserAccountManager sharedInstance].currentUser.credentials.accessToken) ) {
```

Comment:
As long as we're here, can we change the `accessToken` logic to `[SFUserAccountManager sharedInstance].currentUser.credentials.accessToken == nil`?